addons/point_of_sale/models/pos_category.py

```diff
@@ -35,7 +35,7 @@ def _get_hierarchy(self) -> List[str]:
     @api.depends('parent_id')
     def _compute_display_name(self):
         for cat in self:
-            cat.display_name = " / ".join(cat._get_hierarchy())
+            cat.display_name = " / ".join(cat._get_hierarchy()) if cat.name else ""
```

Comment:
Instead of this if expression, I think it's better to modify the _get_hierarchy:
```py
    def _get_hierarchy(self) -> List[str]:
        """ Returns a list representing the hierarchy of the categories. """
        self.ensure_one()
        return (self.parent_id._get_hierarchy() if self.parent_id else []) + [self.name or '']
```

Reply:
Note that the issue is caused by calling the `join` method in the string:
```
" / ".join(cat._get_hierarchy())
```
It's because `cat._get_hierarchy()` return `[False]` which is not automatically casted to array of string.

Reply:
That's true. I changed it